app/controllers/katello/api/v2/products_bulk_actions_controller.rb

```diff
@@ -50,6 +50,21 @@ def sync_products
       respond_for_async :resource => task
     end
 
+    api :PUT, "/products/bulk/verify_checksum", N_("Verify checksum for one or more products")
+    param :ids, Array, :desc => N_("List of product ids"), :required => true
+    def verify_checksum_products
+      repairable_products = @products.syncable
+      repairable_roots = RootRepository.where(:product_id => repairable_products).
+        where(:content_type => ::Katello::Repository::YUM_TYPE).has_url
+
+      repairable_repositories = Katello::Repository.where(:root_id => repairable_roots)
```

Comment:
we may want to only repair the Library instance repos of a product.  Repairing all repos across all content views could take a very long time, and likely contains the same content thats in the Library repo